### DISCUSSION TITLE
Replace r vs or

### DISCUSSION THREAD

**Discussion by Antiever** [BurntSushi/ripgrep#2113] [General] [answered]
Lets say we have a file with content like this:
```
	<strings>
		<string name="EulaHtmlFile" value="EULA.html"/>
		<string name="FunctionalConsentFile" value="GFExperience\FunctionalConsent_${{LocaleName}}.txt"/>
		<string name="RedirectBaseUrl" value="https://www.site.asp"/>
		<string name="URLLanguageIdentifierToken" value="?language=${{AbbrevLangName}}"/>
	</strings>
```
and we do:
```
set file=test.cfg
set RE=.*EulaHtmlFile.*?($\r?\n)
echo !RE!
rg --passthru -i -U "!RE!" "!file!"
```
![cmd_2021-12-23_21-40-01](https://user-images.githubusercontent.com/9666515/147280352-cfbf497d-f045-42b1-87a1-ce287905ffcd.png)

Now add `-o` and get same result. 
Everything looks fine, remove `-o` now add `-r`

`rg --passthru -i -U "!RE!" "!file!" -r ""`

and get:
```
.*EulaHtmlFile.*?($\r?\n)
1-      <strings>
2:              <string name="FunctionalConsentFile" value="GFExperience\FunctionalConsent_${{LocaleName}}.txt"/>
3:              <string name="RedirectBas
3-              <string name="FunctionalConsentFile" value="GFExperience\FunctionalConsent_${{LocaleName}}.txt"/>
4-              <string name="RedirectBaseUrl" value="https://www.site.asp"/>
5-              <string name="URLLanguageIdentifierToken" value="?language=${{AbbrevLangName}}"/>
6-      </strings>
```
mmm... wait, what?

ok, same with `-or`

`rg --passthru -i -U "!RE!" "!file!" -or ""`

```
.*EulaHtmlFile.*?($\r?\n)
1-      <strings>
3-              <string name="FunctionalConsentFile" value="GFExperience\FunctionalConsent_${{LocaleName}}.txt"/>
4-              <string name="RedirectBaseUrl" value="https://www.site.asp"/>
5-              <string name="URLLanguageIdentifierToken" value="?language=${{AbbrevLangName}}"/>
6-      </strings>
```

Now it's ok.
I thought I understand how these parameters work, but now it seems that I do not understand at all how...

**Comment by BurntSushi** [comment, marked as answer]
I believe this is a bug, and probably the same one as #2095.

**Comment by Antiever** [comment]
Honestly, I liked this tool and I would like to use it, so, the question is - are there any assumptions about the timing of fixes?

**Reply by BurntSushi** [reply]
I work on this project in my free time. I don't provide schedules or deadlines. Sorry.

**Reply by Antiever** [reply]
I understand that, that's why I used "assumptions" :) Anyway thx for reply
